### ISSUE TITLE
Suggest: fmap a $ b ==> a <$> b

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#644]
Only if `b` doesn't have a `$` in it, and thus doesn't need bracketing.

**Comment by maartenJacobs** [comment]
@ndmitchell this looks like a good first issue; I like the simplicity of the suggestion. Is this still wanted?

**Comment by shayne-fletcher** [comment]
@maartenJacobs Excuse me for jumping in 😄 I would say, go for it!

**Comment by ndmitchell** [comment]
Yep, would be most welcome!

**Comment by maartenJacobs** [comment]
Before I dig in the wrong direction: `fmap a  $ b ==> a <$> b` is already implemented, right? There's a test in `data/hlint.yml`. Is this improvement for cases like `(fmap (+ 1) . Just) 2 ==> (+ 1) <$> Just 2` then?

**Comment by ndmitchell** [comment]
Hmm, so maybe this already is implemented. Trying the simplest test case:

```
Sample.hs:5:13-24: Suggestion: Use <$>
Found:
  fmap a $ b c
Perhaps:
  a Control.Applicative.<$> b c
```

So it appears it works. Either it was implemented after I created this issue and no one noticed the ticket, or it didn't fire in the example I had because of reasons like old-version, configuration setting etc. Closing this ticket - thanks for looking into it @maartenJacobs !
